ui/src/timeMachine/components/SubmitQueryButton.tsx

```diff
@@ -74,8 +93,24 @@ class SubmitQueryButton extends PureComponent<Props> {
     return ComponentStatus.Default
   }
 
+  private abortController: AbortController
+
   private handleClick = (): void => {
-    this.props.onSubmit()
+    // Optimistic UI, set the state to loading when the event
+    // happens rather than when the event trickles down to execution
+    this.props.setQueryToLoading()
+    // We need to instantiate a new AbortController per request
+    // In order to allow for requests after cancellations:
+    // https://stackoverflow.com/a/56548348/7963795
+    this.abortController = new AbortController()
+    this.props.onSubmit(this.abortController)
+  }
+
+  private handleCancelClick = (): void => {
+    this.props.onNotify(queryCancelRequest())
+    if (this.abortController) {
+      this.abortController.abort()
```

Comment:
I wonder what happens to abortController after you abort it? I'm wondering if you should be setting abortController to null here as well. not sure.

Reply:
That's a great question. Let me find out

Reply:
oooh great question! 🤔

Reply:
I could be wrong, but according to the docs it looks like once an abort controller is aborted, a rejected promise is returned? Although I'm not entirely certain what the means for the instantiated abortController:

https://dom.spec.whatwg.org/#abortcontroller-api-integration

I don't see why setting the value to `null` would hurt